### PR TITLE
Introduce "expect (...) ON x" to be used instead of "expect (...) TO x" on mock objects

### LINKED ISSUE #212: "aMock wasNever called" treats mocks defined inside fixture objects as method calls
We use fixture objects (not traits) and `org.scalatest.fixture._` specs, so mocks are provided into the test like so

```scala
class Fixture {
  val service = mock[Service]
}

// ...

"mySpec" should "test something" in { f: Fixture =>
  f.service wasNever called
}
```

`VerifyMacro` treats `f.service` as a method call, and I get a `NotAMockException`.

I can think of a few ways to solve it:

1. Define a different syntax for verifications on a mock vs a method call. Maybe `aMock wasNever interactedWith`, or even just expose the regular `verifyNoInteractions`?

2. I don't think we can distinguish a method call from a field access using quasiquotes - `q"$a.$b"` matches both if the method has no parens or type args. Here's what I think we could do, however:
    - Method calls with parens is always `Apply` tree
    - Field access or no-paren method calls is always `Select`
    
    We could match against these types specifically, and in the case of `Select`, check if the penultimate identifier is a mock - and if so, treat it as a method call, otherwise as field access.
    To illustrate:
    ```scala
    fixture.aMock wasNever called
    ```
    translates into
    ```scala
    if (mockingDetails(fixture).isMock) verify(fixture, never()).aMock
    else verifyZeroInteractions(fixture.aMock)
    ```

3. (Kinda orthogonal to the above) In Expect DSL, define `on` for mock objects on the right side, and that would be distinct from `to` which only expects method calls, so we'd use
    ```scala
    expect no calls on aMock
    ```
    instead of
    ```scala
    expect no calls to aMock
    ```
    `on` vs `to` would be such a subtle change though so I can imagine errors in these lines be frustrating to diagnose... Maybe a better alternative would be `expect no interactions with aMock`, or again, just expose the vanilla `verifyNoInteractions`/alias it to `expectNoInteractions`

### PR DESCRIPTION
**Pull request by einholen**
Resolves #212 for Expect DSL

### CODE REVIEW

File: macro/src/main/scala/org/mockito/ExpectMacro.scala [hunk 1]
```diff
@@ -4,12 +4,14 @@ import scala.reflect.macros.blackbox
...
+  def callsTo[R](c: blackbox.Context)(stubbedMethodCall: c.Tree)(order: c.Expr[VerifyOrder]): c.Expr[R] = {
     import c.universe._
 
-    val r   = c.Expr[R](transformExpectation(c)(c.macroApplication))
-    val pos = s"${c.enclosingPosition.source.file.name}:${c.enclosingPosition.line}"
-    if (c.settings.contains("mockito-print-expect")) println(pos + " " + show(r.tree))
+    val r = c.Expr[R](transformExpectation(c)(c.macroApplication))
+    if (c.settings.contains("mockito-print-expect")) {
+      val pos = s"${c.enclosingPosition.source.file.name}:${c.enclosingPosition.line}"
+      println(pos + " " + show(r.tree))
```
Comment (ultrasecreth) [inline macro/src/main/scala/org/mockito/ExpectMacro.scala:13]: I like this, we should extract a function and use it everywhere

File: macro/src/main/scala/org/mockito/ExpectMacro.scala [hunk 2]
```diff
@@ -18,30 +20,61 @@ object ExpectMacro extends VerificationMacroTransformer {
...
+              throw new _root_.org.mockito.exceptions.misusing.MissingMethodInvocationException(Seq(
+                "'expect no calls to <?>' requires an argument which is 'a method call on a mock',",
+                "  but [" + $calledPattern + "] is a mock object.",
+                "",
+                "The following would be correct (note the usage of 'calls to' vs 'calls on'):",
+                "    expect no calls to " + $calledPattern + ".bar(*)",
+                "    expect no calls on " + $calledPattern,
+                ""
+              ).mkString("\n"))
+            else ${transformInvocation(c)(q"$obj.$methodOrField", order, q"${c.prefix}.mode")}
```
Comment (ultrasecreth) [inline macro/src/main/scala/org/mockito/ExpectMacro.scala]: Same here, the postfix verification currently uses the standard mockito exception, but I like this one with more info, should be extracted

File: scalatest/src/test/scala/user/org/mockito/PrefixExpectationsTest.scala [hunk 3]
```diff
@@ -46,24 +47,51 @@ class PrefixExpectationsTest extends AnyWordSpec with Matchers with ArgumentMatc
...
+        }
+
+        exception.getMessage shouldBe
+        """'expect no calls to <?>' requires an argument which is 'a method call on a mock',
+            |  but [fixture.org] is a mock object.
+            |
+            |The following would be correct (note the usage of 'calls to' vs 'calls on'):
+            |    expect no calls to fixture.org.bar(*)
+            |    expect no calls on fixture.org
+            |""".stripMargin
```
Comment (ultrasecreth) [inline scalatest/src/test/scala/user/org/mockito/PrefixExpectationsTest.scala]: 🥇

File: scalatest/src/test/scala/user/org/mockito/scalatest/IdiomaticMockitoWithExpectFixtureClassTest.scala [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    "mocked" willBe returned by f.foo.bar("pepe")
+    "mocked" willBe returned by f.foo.baz
+
+    f.foo.bar("pepe") shouldBe "mocked"
+    f.foo.baz shouldBe "mocked"
+
+    a[NeverWantedButInvoked] should be thrownBy {
+      expect no calls to f.foo.bar(*)
+      expect no calls to f.foo.baz
+    }
```
Comment (ultrasecreth) [inline scalatest/src/test/scala/user/org/mockito/scalatest/IdiomaticMockitoWithExpectFixtureClassTest.scala]: this statement should be split as the second line will never be called

**Review by ultrasecreth — CHANGES_REQUESTED** [review]

File: scalatest/src/test/scala/user/org/mockito/scalatest/IdiomaticMockitoWithExpectFixtureClassTest.scala (hunk 4 above)
Comment (ultrasecreth) [reply scalatest/src/test/scala/user/org/mockito/scalatest/IdiomaticMockitoWithExpectFixtureClassTest.scala]: @einholen the rest of the comments are memory helpers for a different PR, but this bit should be fixed

Comment (einholen) [reply scalatest/src/test/scala/user/org/mockito/scalatest/IdiomaticMockitoWithExpectFixtureClassTest.scala]: ah yes great catch! fixed

**Review by ultrasecreth — APPROVED** [review]

File: macro/src/main/scala/org/mockito/ExpectMacro.scala (hunk 1 above)
Comment (einholen) [reply macro/src/main/scala/org/mockito/ExpectMacro.scala:13]: Done in a separate PR: #221 

The two PRs will conflict with each other once merged, I'll resolve whichever comes last

**Pull request merged**